app/src/main/java/io/gnosis/safe/ErrorHandler.kt

```diff
@@ -0,0 +1,106 @@
+package io.gnosis.safe
+
+import android.content.Context
+import androidx.annotation.StringRes
+import io.gnosis.data.repositories.EnsInvalidError
+import io.gnosis.data.repositories.EnsResolutionError
+import io.gnosis.data.repositories.EnsReverseRecordNotSetError
+import io.gnosis.safe.helpers.Offline
+import io.gnosis.safe.ui.safe.add.InvalidName
+import io.gnosis.safe.ui.safe.add.UsedSafeAddress
+import io.gnosis.safe.ui.settings.owner.InvalidSeedPhrase
+import io.gnosis.safe.ui.transactions.details.MismatchingSafeTxHash
+import pm.gnosis.utils.HttpCodes
+import pm.gnosis.utils.exceptions.InvalidAddressException
+import retrofit2.HttpException
+import java.net.ConnectException
+import java.net.SocketTimeoutException
+import java.net.UnknownHostException
+import javax.net.ssl.SSLHandshakeException
+
+enum class Error(
+    val id: Int,
+    @StringRes
+    val reason: Int,
+    @StringRes
+    val howToFix: Int,
+    val trackingRequired: Boolean = false
+
+) {
+    // Network-related errors
+    ERROR_101(101, R.string.error_network_no_internet_reason, R.string.error_network_no_internet_fix),
+    ERROR_102(102, R.string.error_network_ssl_reason, R.string.error_network_ssl_fix),
+    ERROR_103(103, R.string.error_network_timeout_reason, R.string.error_network_timeout_fix) {
+        override fun message(context: Context): String {
+            return "${context.getString(reason, 30)}. ${context.getString(howToFix)}. (${context.getString(R.string.error_id, id)})"
+        }
+    },
+    ERROR_104(104, R.string.error_network_unknown_host_reason, R.string.error_network_unknown_host_fix),
+    ERROR_400(400, R.string.error_network_request_reason, R.string.error_network_request_fix),
+    ERROR_401(401, R.string.error_network_not_authorized_reason, R.string.error_network_not_authorized_fix),
+    ERROR_403(403, R.string.error_network_not_authorized_reason, R.string.error_network_not_authorized_fix),
+    ERROR_500(500, R.string.error_network_server_side_reason, R.string.error_network_server_side_fix),
+
+    // Common client errors
+    ERROR_1101(1101, R.string.error_client_safe_duplicate_reason, R.string.error_client_safe_duplicate_fix),
+    ERROR_1102(1102, R.string.error_client_address_invalid_reason, R.string.error_client_address_invalid_fix),
+    ERROR_1103(1103, R.string.error_client_seed_phrase_reason, R.string.error_client_seed_phrase_fix),
+    ERROR_1104(1104, R.string.error_client_tx_hash_validation_reason, R.string.error_client_tx_hash_validation_fix),
+    ERROR_1105(1105, R.string.error_client_unsupported_implementation_copy_reason, R.string.error_client_unsupported_implementation_copy_fix),
```

Comment:
This is not used. Do we need it later, or should it be used already?

Reply:
we might want to make safe validation more strict and allow adding safes that have min required version of the implementation copy.

Reply:
I'll add this to the next refinement. We have to adjust our safe validation